fairseq/utils.py

```diff
@@ -146,13 +146,16 @@ def load_ensemble_for_inference(filenames, task, model_arg_overrides=None):
         state = torch.load(filename, map_location=lambda s, l: default_restore_location(s, 'cpu'))
         state = _upgrade_state_dict(state)
         states.append(state)
-    args = states[0]['args']
+    args_list = []
+    for state in states:
+        args_list.append(state['args'])
     if model_arg_overrides is not None:
-        args = _override_model_args(args, model_arg_overrides)
+        for args in args_list:
+            args = _override_model_args(args, model_arg_overrides)
```

Comment:
This doesn't quite work. It will reassign the locally-scoped args to the overridden version, but this will not be reflected in args_list (except that args is passed into _override_model_args by reference, so it might get changed in-place).

But anyway, let's get rid of args_list and this entire block above. We can do it inline below instead.